### PR TITLE
Method DR.SC

### LINKED ISSUE #21: method_DR-SC
**Update metadata:**
https://docs.google.com/spreadsheets/d/1QCeAF4yQG4bhZSGPQwwVBj_XF7ADY_2mK5xivAIfHsc/edit#gid=0

**Details of method (publication, GitHub):**
https://academic.oup.com/nar/article/50/12/e72/6555431
https://github.com/feiyoung/DR.SC

### PR DESCRIPTION
**Pull request by shdam**
DR.SC implemented (Issue #21)

Package is on CRAN, but not available for R 4.3.2, which is the version where SpatialExperiment. Therefore, I had to get it from GitHub.
The environment can be created with drsc_env.sh.

It was not possible to use the neighbors for this, as their neighbor function is hardcoded to be run. 
Seurat needs a counts and logcounts assay, so I just added the counts matrix as both.

### CODE REVIEW

**Review by niklasmueboe — CHANGES_REQUESTED** [review]

File: method/DRSC/DRSC.r [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+# You can use the data as SpatialExperiment
+spe <- get_SpatialExperiment(feature_file = feature_file, observation_file = observation_file,
+                                    coord_file = coord_file,matrix_file = matrix_file)
+
+
+## Your code goes here
+seurat_obj <- as.Seurat(spe)
+seurat_obj <- NormalizeData(seurat_obj, verbose=FALSE)
+seurat_obj <- FindSVGs(seurat_obj, verbose=FALSE)
```
[the remark this answers is not in the document]
Comment (shdam) [reply method/DRSC/DRSC.r:170]: I now assign the features as `seurat_obj[["originalexp"]]@var.features <- rownames(rowData(spe))`, as they are subset to the selected when generating spe:
`spe <- spe[as.logical(rowData(spe)$selected), ]`

How would I add the normalized data? Is that a separate file input?

[the remark this answers is not in the document]
Comment (niklasmueboe) [reply method/DRSC/DRSC.r:170]: Assume that the matrix_file is preprocessed as needed

**Comment by shdam** [comment]
All comments addressed :)

File: method/DRSC/DRSC.yml
[the remark this answers is not in the document]
Comment (shdam) [reply method/DRSC/DRSC.yml:1]: I removed spatialexperiment as a dependency and got it to work with the CRAN version of the package.

File: method/DRSC/DRSC.r (hunk 1 above)
[the remark this answers is not in the document]
Comment (shdam) [reply method/DRSC/DRSC.r]: Got it. Wrote it as "normalized counts" in the sheet.

**Review by niklasmueboe — CHANGES_REQUESTED** [review]

**Review by niklasmueboe — APPROVED** [review]

File: method/DRSC/DRSC.r [hunk 2]
```diff
@@ -129,13 +128,12 @@ get_SpatialExperiment <- function(
...
-  spe <- SpatialExperiment::SpatialExperiment(
-    rowData = rowData, colData = colData, spatialCoords = coordinates
-  )
+    spe <- SingleCellExperiment::SingleCellExperiment(
+    rowData = rowData, colData = colData, metadata = list("spatialCoords" = coordinates))
 
   if (!is.na(matrix_file)) {
     assay(spe, assay_name, withDimnames = FALSE) <- as(Matrix::t(Matrix::readMM(matrix_file)), "CsparseMatrix")
-    assay(spe, "logcounts", withDimnames = FALSE) <- as(Matrix::t(Matrix::readMM(matrix_file)), "CsparseMatrix")
+    assay(spe, "logcounts", withDimnames = FALSE) <- log1p(as(Matrix::t(Matrix::readMM(matrix_file)), "CsparseMatrix"))
```
[the remark this answers is not in the document]
Comment (niklasmueboe) [reply method/DRSC/DRSC.r]: Which of the assays will be used. Log counts? or counts?

[the remark this answers is not in the document]
Comment (shdam) [reply method/DRSC/DRSC.r]: Good point! I am not a big Seurat user - but I checked and it is the logcounts slot that is used. It gets renamed to "data" when running as.seurat, and the package uses the "data" slot. So it did make a difference!

**Pull request merged**